homeassistant/components/logbook.py

```diff
@@ -97,6 +118,14 @@ def humanify(events):
             if event.event_type == EVENT_STATE_CHANGED:
                 entity_id = event.data['entity_id']
 
+                # This should be rafactored so that the SQL query does
+                # the filtering, possibly by adding an option entity_id
+                # field to events
+                if (filter_entity_id is not None and
```

Comment:
No need for the check `filter_entity_id is not None`. If it is it will never be equal to entity_id.